platform/mellanox/mlnx-platform-api/sonic_platform/eeprom.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+
+#############################################################################
+# Mellanox
+#
+# Module contains an implementation of SONiC Platform Base API and
+# provides the eeprom information which are available in the platform
+#
+#############################################################################
+import exceptions
+import os
+import sys
+import re
+from cStringIO import StringIO
+
+try:
+    from sonic_platform_base.sonic_eeprom import eeprom_tlvinfo
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+
+#
+# CACHE_XXX stuffs are supposted to be moved to the base classes
+# since they are common for all vendors
+# they are defined in decode-syseeprom which might be removed in the future
+# currently we just copy them here
+#
+CACHE_ROOT = '/var/cache/sonic/decode-syseeprom'
+CACHE_FILE = 'syseeprom_cache'
+
+#
+# this is mlnx-specific
+# should this be moved to chass.py or here, which better?
+#
+EEPROM_SYMLINK = "/var/run/hw-management/eeprom/vpd_info"
+
+class Eeprom(eeprom_tlvinfo.TlvInfoDecoder):
+    RETRIES = 5
+    EEPROM_DECODE_HEADLINES = 6
+    EEPROM_DECODE_MAXITEM = 3
+    EEPROM_DECODE_OFFSET = 0
+    EEPROM_DECODE_CONTENT = 2
+
+    def __init__(self):
+        for attempt in range(self.RETRIES):
```

Comment:
why do we need to keep retries? usually 3 retries is more general approach. why 5?

Reply:
This is a piece of code I ported from legacy code. 
But I agree with you, to retry for 3 times is a common practice. I will fix it.